bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BServiceType.java

```diff
@@ -49,26 +50,26 @@ public void setResourceFunctions(ResourceFunctionType[] resourceFunctions) {
      * @return array of remote functions
      */
     @Override
-    public MemberFunctionType[] getRemoteFunctions() {
-        if (remoteFunctions == null) {
-            MemberFunctionType[] funcs = getRemoteFunctions(getAttachedFunctions());
+    public RemoteMethodType[] getRemoteMethods() {
+        if (remoteMethods == null) {
```

Comment:
Not related to this PR, but can't we do this when setting the methods itself, by overriding `setAttachedFunctions` maybe? We can probably remove the synchronized block and make `remoteMethods` a non-volatile variable then, right?